sympy/solvers/ode/single.py

```diff
@@ -55,9 +61,10 @@ class SingleODEProblem:
     sym = None  # type: Symbol
     _order = None
 
-    def __init__(self, eq, func, sym):
+    def __init__(self, eq, func, sym, eq_orig=None):
         self.eq = eq
         self.func = func
+        self.eq_orig = eq_orig
```

Comment:
Is `eq_orig` ever different from `eq` here?

Reply:
It is not different in these cases but some other cases might need it as mentioned in the documentation of `classify_ode`.

Reply:
Here in this branch eq argument to `SingleODEProblem` is what is called `eq_orig` in `classify_ode`? That is deliberate because I want `eq` to be the original equation.

Reply:
> Here in this branch eq argument to `SingleODEProblem` is what is called `eq_orig` in `classify_ode`?

The `eq_orig` is different from eq after the function `_preprocess` is applied here.
https://github.com/oscarbenjamin/sympy/blob/2ee06f3b3359ef3986878825e403c806b2b0c6ef/sympy/solvers/ode/ode.py#L975-L978

Reply:
I know. That's why I deliberately used `eq_orig` in the argument to SingleODEProblem.

I would like this logic to got into `SingleODEProblem` somehow. I think that it should have properties for getting different forms of the equation e.g.: `eq_preprocessed` or `eq_expanded`. The `eq` attribute should be the original ODE with no processing.

In the transition we should continue to support the `prep` argument to `dsolve` so I suggest adding that as an optional kwyword-only parameter to SingleODEProblem like
```
class SingleODEProblem:
    def __init__(self, eq, func, sym, *, prep=True):
        ...
```
Then in `classify_ode` we should do
```
problem = SingleODEProblem(eq_orig, func, sym, prep=prep)
```

Reply:
> I would like this logic to got into `SingleODEProblem` somehow. I think that it should have properties for getting different forms of the equation e.g.: `eq_preprocessed` or `eq_expanded`. The `eq` attribute should be the original ODE with no processing.

Sorry for the misunderstanding. As of now, I have added the `eq_expanded` property in my latest commit. Should I also add this feature in the same PR or is this supposed to be added along with more refactored methods in a different PR?

Reply:
You can just make the changes in new commits and push to this PR.